File: src/yellow/ui/buttons/dialogs/YellowWeaponSwitchDialog.java
package yellow.ui.buttons.dialogs;

import arc.util.*;
import arc.util.Log.*;
import arc.scene.ui.*;
import mindustry.*;
import mindustry.gen.*;
import mindustry.ui.dialogs.*;
import mindustry.entities.units.*;
import yellow.type.*;
import yellow.content.*;
import yellow.weapons.*;
import yellow.entities.units.*;
import yellow.interactions.*;
import yellow.interactions.ui.*;

import static yellow.interactions.Responses.*;

public class YellowWeaponSwitchDialog extends BaseDialog{
    private static DialogueBox box = new DialogueBox();
    
    public YellowWeaponSwitchDialog(){
        super("Weapon Switch");
        
        cont.add("Weapon Switch").row();
        addCloseButton();
        
        for(int i = 0; i < YellowUnitTypes.yellow.weapons.size; i++){
            final int id = i;
            
            cont.check(((NameableWeapon) YellowUnitTypes.yellow.weapons.get(id)).displayName + " Lock", false, it -> {
                Unit unit = Vars.player.unit();
                DisableableWeaponMount m = ((DisableableWeaponMount) unit.mounts[id]);
                m.disabled = it;
            }).row();
        };
        
        cont.button("Request Leave", () -> {
            box.dialogueStart(leaveResponse, new Runnable[]{
                () -> Log.info("To be tested...")
            }, new int[]{2});
        }).row();
    };
}

File: src/yellow/ui/buttons/dialogs/YellowWeaponSwitchDialog.kt
package yellow.ui.buttons.dialogs

import com.github.mnemotechnician.mkui.extensions.dsl.textButton
import mindustry.Vars
import mindustry.entities.units.WeaponMount
import mindustry.ui.dialogs.BaseDialog
import yellow.Yellow
import yellow.entities.units.DisableableWeaponMount
import yellow.entities.units.entity.YellowUnitEntity
import yellow.type.*

open class YellowWeaponSwitchDialog: BaseDialog("Weapon Switch") {
    init {
        addCloseButton()
    }

    fun show(weapon: Array<WeaponMount>) = show(weapon, null)

    fun show(weapon: Array<WeaponMount>, unit: YellowUnitEntity?) {
        cont.clear()
        weapon.forEach {
            if(it !is DisableableWeaponMount) return
            if(!(it.weapon as DisableableWeapon).mirroredVersion){
                cont.check((it.weapon as NameableWeapon).displayName, it.enabled) {a: Boolean ->
                    it.enabled = a
                    if(it.enabled) it.enabled() else it.disabled()
                }
                cont.textButton("?") {
                    Yellow.weaponInfo.show(it.weapon as NameableWeapon)
                }.row()
            }else{
                cont.check((it.weapon as NameableWeapon).displayName, it.enabled) {a: Boolean ->
                    it.enabled = a
                    if(it.enabled) it.enabled() else it.disabled()
                }.row()
            }
        }
        if(!Vars.mobile && unit != null) cont.check("@sentryidle", unit.forceIdle){a: Boolean ->
            unit.forceIdle = a
        }

        super.show()
    }
}
